app/main.py

```diff
@@ -1,2 +1,43 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 0.67
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._get_index(key)
+        if self.table[index] is None:
+            self.size += 1
+            self.table[index] = (key, hash(key), value)
+        elif self.table[index] is not None:
+            self.table[index] = (key, hash(key), value)
```

Comment:
Make it DRY. Move `self.table[index] = (key, hash(key), value)` out of condition